pkg/cloud-provider/cloudapi/azure/azure_test.go

```diff
@@ -244,7 +261,160 @@ var _ = Describe("Azure", func() {
 			filters := serviceConfig.(*computeServiceConfig).computeFilters[selector.Name]
 			Expect(filters).To(Equal(expectedQueryStrs))
 		})
+
+		It("Should match expected filter - single VMID only match", func() {
```

Comment:
Try to add all combination. 
1) VNETID match
2) VMID match
3) VM Name Match
4) VNETID + VMID
5) VNETID + VMName
6) Update and rollback. Like start with VNETID, then add VM Match and then remove VM Match... and so on

Reply:
Here is the problem now, seems like our RemoveSelector is not completed, so we cannot do the rollback now